android/build.gradle

```diff
@@ -23,6 +30,46 @@ android {
     defaultConfig {
         minSdkVersion safeExtGet('minSdkVersion', 18)
         targetSdkVersion safeExtGet('targetSdkVersion', 30)
+        buildConfigField "boolean", "IS_NEW_ARCHITECTURE_ENABLED", isNewArchitectureEnabled().toString()
+
+        if (isNewArchitectureEnabled()) {
+            var appProject = rootProject.allprojects.find { it.plugins.hasPlugin('com.android.application') }
+            externalNativeBuild {
+                ndkBuild {
+                    arguments "APP_PLATFORM=android-21",
+                            "APP_STL=c++_shared",
+                            "NDK_TOOLCHAIN_VERSION=clang",
+                            "GENERATED_SRC_DIR=${appProject.buildDir}/generated/source",
+                            "PROJECT_BUILD_DIR=${appProject.buildDir}",
+                            "REACT_ANDROID_DIR=${appProject.rootDir}/../node_modules/react-native/ReactAndroid",
```

Comment:
This is an unsafe assumption about where react-native package is located that works with this project, but breaks mono-repos (that we use at Klarna). See this [gradle file](https://github.com/oblador/react-native-performance/blob/master/packages/react-native-performance/android/build.gradle#L15-L28) for an example on how that can be done in a more dynamic fashion. I've discussed this with Meta but no better solution in place than this from what I know.